packages/devtools_app/lib/src/memory/memory_instance_tree_view.dart

```diff
@@ -0,0 +1,166 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+import '../auto_dispose_mixin.dart';
+import '../table.dart';
+import '../table_data.dart';
+import '../utils.dart';
+import 'memory_controller.dart';
+import 'memory_snapshot_models.dart';
+
+class InstanceTreeView extends StatefulWidget {
+  @override
+  InstanceTreeViewState createState() => InstanceTreeViewState();
+}
+
+/// Table of the fields of an instance (type, name and value).
+class InstanceTreeViewState extends State<InstanceTreeView>
+    with AutoDisposeMixin {
+  MemoryController controller;
+
+  final TreeColumnData<FieldReference> treeColumn = _FieldTypeColumn();
+  final List<ColumnData<FieldReference>> columns = [];
+
+  @override
+  void initState() {
+    setupColumns();
+
+    super.initState();
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+
+    final newController = Provider.of<MemoryController>(context);
+    if (newController == controller) return;
+    controller = newController;
+
+    cancel();
+
+    // TODO(terry): setState should be called to set our state not change the
+    //              controller. Have other ValueListenables on controller to
+    //              listen to, so we don't need the setState calls.
+    // Update the chart when the memorySource changes.
+    addAutoDisposeListener(controller.selectedSnapshotNotifier, () {
+      setState(() {
+        controller.computeAllLibraries(rebuild: true);
+      });
+    });
+  }
+
+  void setupColumns() {
```

Comment:
if setupColumns is only called in initState, consider inlining it in initState.

Reply:
Moved all setupColumns to initState.